### PR TITLE
fix(openapi): endpoint method description

### PR DESCRIPTION
**Pull request by pcanto-hopeit**
**PROBLEM**

**SOLUTION**

**REVIEW**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -59,8 +59,11 @@ dist-only: clean
 	cd engine && \
 	python setup.py sdist bdist_wheel
 
-pypi: dist
-	twine upload --repository pypi dist/*
+pypi_test: 
+	python -m twine upload --repository testpypi engine/dist/*
+
+#pypi: dist
+#	twine upload --repository pypi dist/*
```
Comment (leosmerling-hopeit) [inline Makefile:66]: Remove this change from PR

File: engine/README.md [hunk 2]
```diff
@@ -28,10 +28,13 @@
...
+    --host: server host address or name, default is --host=0.0.0.0
     --port: indicates to listen on another port number, default is --port=8020
-    --path: indicates to listen in a unix socket path, default is disabled
+    --path: indicates to listen in a unix socket path, default is disabled    
     --start-streams: indicates to auomatically start events of type STREAM when starting server
-    --config-files: comma-separated file of config files to load, starting with server config, then plugins, then apps
+    
 ```
 
 - Example starting a single app that depends on plugins:
```
Comment (leosmerling-hopeit) [inline engine/README.md:40]: Remove this file from PR, make another.

File: engine/setup.py [hunk 3]
```diff
@@ -2,11 +2,36 @@
 
 setuptools.setup(
     name="hopeit.engine",
-    version="0.1.2",
+    version="0.1.3",
     description="Hopeit.py Engine",
+    long_description=open('README.md').read(),
```
Comment (leosmerling-hopeit) [inline engine/setup.py:7]: Remove this changes from PR.

File: engine/src/hopeit/app/api.py [hunk 4]
```diff
@@ -96,6 +96,22 @@ def _payload_description(arg: PayloadDef) -> str:
     return str(arg)
 
 
+def _method_summay(title, module) -> str:
```
Comment (leosmerling-hopeit) [inline engine/src/hopeit/app/api.py:99]: typo

File: engine/src/hopeit/app/api.py [hunk 5]
```diff
@@ -96,6 +96,22 @@ def _payload_description(arg: PayloadDef) -> str:
...
+    ds = inspect.getdoc(module)
+    if title is not None:
+        return title
+    if isinstance(ds, str):
+        return ds.split("\n")[0]
+    return ""
+
+
+def _method_description(title, module) -> str:
```
Comment (leosmerling-hopeit) [inline engine/src/hopeit/app/api.py:108]: where title comes from?

**Review by leosmerling-hopeit — CHANGES_REQUESTED** [review]

**Comment by leosmerling-hopeit** [comment]
@pcanto-hopeit remember to increase engine & simple_example version.

File: engine/src/hopeit/app/api.py [hunk 6]
```diff
@@ -96,6 +96,22 @@ def _payload_description(arg: PayloadDef) -> str:
...
+    ds = inspect.getdoc(module)
+    if ds is not None and ds.count('\n') > 1:
```
Comment (leosmerling-hopeit) [inline engine/src/hopeit/app/api.py:110]: Make split at the beginning only once, and remove count check.

File: engine/src/hopeit/app/api.py [hunk 7]
```diff
@@ -96,6 +96,22 @@ def _payload_description(arg: PayloadDef) -> str:
...
+        return ds.split("\n", 2)[2]
```
Comment (leosmerling-hopeit) [inline engine/src/hopeit/app/api.py:111]: Add unit tests for these cases.

File: engine/src/hopeit/app/api.py (hunk 5 above)
Comment (pcanto-hopeit) [reply engine/src/hopeit/app/api.py:108]: it gets from event_api parameter

File: engine/src/hopeit/app/api.py (hunk 7 above)
Comment (pcanto-hopeit) [reply engine/src/hopeit/app/api.py]: done on commit https://github.com/hopeit-git/hopeit.engine/pull/23/commits/4db72834126369b5482702b288f3a9d3494ec45d

File: engine/src/hopeit/app/api.py [hunk 8]
```diff
@@ -152,13 +172,15 @@ def _event_api(
...
 def event_api(title: Optional[str] = None,
+              description: Optional[str] = None,
               payload: Optional[PayloadDef] = None,
               query_args: Optional[List[ArgDef]] = None,
               responses: Optional[Dict[int, PayloadDef]] = None) -> Callable[..., dict]:
     """
     Provides a convenient way to define Open API specification using Python types for a given app event
     implementation module.
-
+    :param title: An optional, string summary. If not provides it gets from the doc_string definition.
```
Comment (leosmerling-hopeit) [inline engine/src/hopeit/app/api.py:182]: "If not provided will be taken from module docstring"

File: engine/src/hopeit/app/api.py [hunk 9]
```diff
@@ -152,13 +172,15 @@ def _event_api(
...
+    :param description: An optional, string description. If not provides it gets from the doc_string definition.
```
Comment (leosmerling-hopeit) [inline engine/src/hopeit/app/api.py:183]: "If not provided witll be taken from ..."

File: engine/test/integration/server/test_it_web.py [hunk 10]
```diff
@@ -1,18 +1,19 @@
...
+
 from pytest_aiohttp import aiohttp_server, aiohttp_client  # type: ignore  # noqa: F401
 
 import hopeit.server.web
 from hopeit.server import api
 from hopeit.server.web import start_server, stop_server, start_app
 from hopeit.server.streams import StreamManager
 
-from mock_engine import MockStreamManager, MockEventHandler
+from mock_engine import MockStreamManager, MockEventHandler  # type: ignore
```
Comment (leosmerling-hopeit) [inline engine/test/integration/server/test_it_web.py:16]: Why type: ignore? Maybe some methods are missing to be implemented in MockStreamManager or MockEventHandler?

File: engine/test/integration/toolkit/test_it_auth.py [hunk 11]
```diff
@@ -13,7 +13,7 @@
 from hopeit.toolkit import auth
 from hopeit.server.config import AuthConfig, AuthType
 
-from mock_app import mock_app_config  # noqa: F401
+from mock_app import mock_app_config  # type: ignore  # noqa: F401
```
Comment (leosmerling-hopeit) [inline engine/test/integration/toolkit/test_it_auth.py:16]: Why type: ignore?

File: engine/test/mock_app/__init__.py [hunk 12]
```diff
@@ -435,7 +438,8 @@ def mock_api_spec():
                             "type": "string"
                         }
                     },
-                    "description": "MockData object"
+                    "description": "MockData object",
+                    "x-module-name": "mock_app"
```
Comment (leosmerling-hopeit) [inline engine/test/mock_app/__init__.py:442]: Where this x-module-name comes from?

File: engine/test/mock_app/mock_app_api_get.py [hunk 13]
```diff
@@ -5,13 +5,14 @@
 
 from hopeit.app.logger import app_extra_logger
 from hopeit.app.context import EventContext
-from mock_app import MockData
+from mock_app import MockData  # type: ignore
```
Comment (leosmerling-hopeit) [inline engine/test/mock_app/mock_app_api_get.py:8]: Why type: ignore?

File: engine/test/mock_app/mock_app_api_get_list.py [hunk 14]
```diff
@@ -1,12 +1,14 @@
 """
 NO TITLE HERE, Title in __api__
+----------------
+Description of Test app api list
 """
```
Comment (leosmerling-hopeit) [inline engine/test/mock_app/mock_app_api_get_list.py:5]: What about the case where there's only a description here with no delimiters and title is on event_api? Is it covered?

File: engine/test/mock_app/mock_app_api_post.py [hunk 15]
```diff
@@ -7,11 +7,12 @@
 
 __steps__ = ['entry_point']
 
-from mock_app import MockData
+from mock_app import MockData  # type: ignore
```
Comment (leosmerling-hopeit) [inline engine/test/mock_app/mock_app_api_post.py:10]: Why type: ignore?

File: engine/test/unit/app/test_app_api.py [hunk 16]
```diff
@@ -5,7 +5,7 @@
 import hopeit.app.api as api
 from hopeit.app.config import EventType
 from hopeit.server.api import APIError
-from mock_app import mock_app_api_get, MockData, mock_app_api_post, mock_app_api_get_list
+from mock_app import mock_app_api_get, MockData, mock_app_api_post, mock_app_api_get_list  # type: ignore
```
Comment (leosmerling-hopeit) [inline engine/test/unit/app/test_app_api.py:8]: Why type: ignore? (resolve if answered before)

File: engine/test/unit/server/test_api.py [hunk 17]
```diff
@@ -5,7 +5,7 @@
 
 import pytest
 from aiohttp import web
-from aiohttp_swagger3.swagger_route import SwaggerRoute
+from aiohttp_swagger3.swagger_route import SwaggerRoute  # type: ignore
```
Comment (leosmerling-hopeit) [inline engine/test/unit/server/test_api.py:8]: Why type:ignore here?

File: engine/test/unit/server/test_compression.py [hunk 18]
```diff
@@ -4,7 +4,7 @@
 import gzip
 import bz2
 import lzma
-import lz4.frame
+import lz4.frame  # type: ignore
```
Comment (leosmerling-hopeit) [inline engine/test/unit/server/test_compression.py:7]: And why here? It looks like libs are not updating its type specs with mypy versions, but we should keep the usage of type:ignore as reduced as possible, so please confirm the reason on each different case. You are not supossed to introduce changes to files not affected by the functionllity so I would recommend openining a new PR to deal with mypy problems for existing/untouched code.

File: engine/test/unit/server/test_engine.py [hunk 19]
```diff
@@ -9,7 +9,7 @@
 from hopeit.dataobjects import DataObject
 from hopeit.app.config import AppConfig
 from hopeit.server.engine import AppEngine
-from mock_engine import MockEventHandler, MockStreamManager
+from mock_engine import MockEventHandler, MockStreamManager  # type: ignore
```
Comment (leosmerling-hopeit) [inline engine/test/unit/server/test_engine.py:12]: Why type: ignore?

File: engine/test/unit/server/test_events.py [hunk 20]
```diff
@@ -3,7 +3,7 @@
 from hopeit.app.context import EventContext, PostprocessHook
 from hopeit.server.config import AuthType
 from hopeit.server.events import EventHandler
-from mock_engine import MockStreamManager
+from mock_engine import MockStreamManager  # type: ignore
```
Comment (leosmerling-hopeit) [inline engine/test/unit/server/test_events.py:6]: This file should not be part of this PR.

File: engine/test/unit/server/test_server.py [hunk 21]
```diff
@@ -5,7 +5,7 @@
 
 from mock_app import mock_app_config  # type: ignore  # noqa: F401
 from mock_plugin import mock_plugin_config  # type: ignore  # noqa: F401
-from mock_engine import MockAppEngine
+from mock_engine import MockAppEngine  # type: ignore
```
Comment (leosmerling-hopeit) [inline engine/test/unit/server/test_server.py:8]: This file should not be part of this PR.

File: engine/test/unit/server/test_streams.py [hunk 22]
```diff
@@ -4,8 +4,8 @@
 from datetime import datetime, timezone
 from dataclasses import dataclass
 
-from hopeit.app.config import Compression, Serialization
-from mock_engine import MockStreamManager, MockEventHandler
+from hopeit.app.config import Compression, Serialization   # type: ignore
+from mock_engine import MockStreamManager, MockEventHandler  # type: ignore
 from hopeit.dataobjects import dataobject
```
Comment (leosmerling-hopeit) [inline engine/test/unit/server/test_streams.py:9]: This file should not be part of this PR.

**Review by leosmerling-hopeit — CHANGES_REQUESTED** [review]
-Minor changes requested
-Remove files that are not related to this PR
-Revert usage of type:ignore when possible or explain reason/investigate further.

File: engine/test/unit/server/test_api.py (hunk 17 above)
Comment (pcanto-hopeit) [reply engine/test/unit/server/test_api.py:8]: stubs are not provided for that module
